sdks/java/core/src/main/java/org/apache/beam/sdk/testing/TestPipeline.java

```diff
@@ -341,7 +348,12 @@ public PipelineResult run(PipelineOptions options) {
     final PipelineResult pipelineResult;
     try {
       enforcement.get().beforePipelineExecution();
-      pipelineResult = super.run(options);
```

Comment:
I think this also needs to serialize/deserialize the graph, it isn't already happening. Specifically, we saw problems when serializing a graph containing a Nested Value Provider.

Reply:
It's already happening as part of direct runner I think (and other runners of course).

Reply:
To confirm, if you intentionally break AvroIO to use the anonymous NestedValueProvider, the test still catches it, correct?

(Just want to verify we aren't losing regression coverage)